### ISSUE TITLE
@typescript-eslint/internal/plugin-test-formatting too strict about snapshot indentation

### ISSUE THREAD

**Issue by Josh-Cena** [typescript-eslint/typescript-eslint#4911]
Complained in https://github.com/typescript-eslint/typescript-eslint/pull/4901#discussion_r864456822

Take this code for example (from that PR):

```ts
{
      code: `
class Data {
  member<T extends unknown>() {}
}
      `,
      errors: [
        {
          data: { constraint: 'unknown', name: 'T' },
          messageId: 'unnecessaryConstraint',
          endColumn: 27,
          column: 10,
          line: 3,
          suggestions: [
            {
              messageId: 'removeUnnecessaryConstraint',
              output: `
class Data {
  member<T>() {}
}
              `,
            },
          ],
        },
      ],
    },
```

The test fails, suggesting that:

```
Expected value to strictly be equal to:
      "
    class Data {
      member<T>() {}
    }
                  "
    Received:
      "
    class Data {
      member<T>() {}
    }
          "
```

Okay, let's try another way by indenting everything—maybe it will be dedented? (Spoiler: I don't know how the snapshot works.)

```ts
{
      code: `
        class Data {
          member<T extends unknown>() {}
        }
      `,
      errors: [
        {
          data: { constraint: 'unknown', name: 'T' },
          messageId: 'unnecessaryConstraint',
          endColumn: 35,
          column: 18,
          line: 3,
          suggestions: [
            {
              messageId: 'removeUnnecessaryConstraint',
              output: `
                class Data {
                  member<T>() {}
                }
              `,
            },
          ],
        },
      ],
    },
```

```
    - Expected
    + Received


    -                 class Data {
    -                   member<T>() {}
    -                 }
    -               
    +         class Data {
    +           member<T>() {}
    +         }
    +       
```

o_o

It turns out the only way to work around it is by either fighting the linter and adding ignore comments, or by using `.trimEnd()` to each snapshot.

I think it's fine for the snapshots to be formatted nicely, so what about a sanitation step that removes all extraneous indentation, or removes trailing spaces (like `.trimEnd()`)? I think ESLint's tests does that.

**Comment by bradzacher** [comment]
yeah in general we just add `trimEnd` to these cases.
The difficult thing about adding `trimEnd` at a framework level is that we don't know if that whitespace actually matters.
For example - what happens if a fixer intentionally fixes `const x = 1;` to `const x = 1; `? Then it would be impossible to test the fixer!

**Comment by Josh-Cena** [comment]
> what happens if a fixer intentionally fixes `const x = 1;` to `const x = 1; `?

It shouldn't matter, though. That should fall under ESLint's principle of "fixers shouldn't care about code style". At worst it would mean the user needs to format their code again after fixing, which they probably always do anyways.

**Comment by JoshuaKGoldberg** [comment]
Following up on this issue: I agree, it's annoying to have to battle with Prettier on test `output` properties. If a fixer does create some bad whitespace situation, we'll see it explicitly in the test file.
